### ISSUE TITLE
Type String is not determined when port numbers in FromPort and ToPort are used with single (') or double (") quotes

### ISSUE THREAD

**Issue by tblinux** [aws-cloudformation/cfn-lint#2652]
### CloudFormation Lint Version

0.76.1

### What operating system are you using?

Mac

### Describe the bug

The type string is not determined when the port number is used with single (') or double ("") quotes in FromPort and ToPort in security group rules for AWS::EC2::SecurityGroup resource.

### Expected behavior

When single (') or double (") quotes are used in security group rules for AWS::EC2::SecurityGroup resource in the AWS CloudFormation template. 

**If string is used in FromPort, it should throw an error like below**

"E3012 Property Resources/DemoSecurityGroup/Properties/SecurityGroupIngress/0/FromPort should be of type Integer"

**If string is used in ToPort, it should throw an error like below**
"E3012 Property Resources/DemoSecurityGroup/Properties/SecurityGroupIngress/0/ToPort should be of type Integer"

### Reproduction template

```
AWSTemplateFormatVersion: '2010-09-09'
Resources:
  DemoSecurityGroup:
    Type: 'AWS::EC2::SecurityGroup'
    Properties:
      VpcId: vpc-000000 #dummy vpc id
      GroupDescription: SG to allow SSH access via port 22
      SecurityGroupIngress:
        - Description: Allowing port 22 for everyone
          IpProtocol: tcp
          FromPort: '22'
          ToPort: '22'
          CidrIp: "0.0.0.0/0"
```

**Comment by kddejong** [comment]
This is a long running discussion.  By default we act like the service does in which we accept either a string or an integer.  You can force the rule to act more like the spec by adding the parameter `-X E3012:strict=true` to your CLI or by adding it to your `.cfnlintrc` file.  This will cause E3012 to fail as you have asked it to.

That being said one of the issues is that certain resources will have a spec that says string when the value can only be an integer, etc.

**Comment by tblinux** [comment]
I tested with CLI option `-x E3012:strict=true` and it is working as expected. 

As per the AWS documentation [reference](https://docs.aws.amazon.com/AWSCloudFormation/latest/UserGuide/aws-properties-ec2-security-group-rule-1.html#cfn-ec2-security-group-rule-fromport) It should be integer but yes it's correct that, Service does accept single (') and double quotes (") with port number. 

I faced this first time when cfn-lint and security scanning tools both couldn't catch this but rule was actually non-compliant.

**Comment by kddejong** [comment]
Going to close this as it is working as planned. There is a way to configure this check to be more strict.
